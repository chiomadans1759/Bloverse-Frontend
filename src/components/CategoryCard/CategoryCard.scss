@import 'sass-rem';

.category {
  font-size: rem(12px);
}

.category-title {
  font-size: rem(14px) !important;
}

.edit-icon {
  font-size: rem(20px);
  color: #096DD9;
}
