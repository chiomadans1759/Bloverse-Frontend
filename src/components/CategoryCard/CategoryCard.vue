<template>
  <main class="category-card p-4 p-sm-3" :class="{ card: !hideCardClass }">
    <div class="header d-flex justify-content-between ">
      <h5><strong>Interest</strong></h5>
      <span class="edit-icon"><i class="fal fa-edit mr-2"></i></span>
    </div>
    <div class="mt-1">
      <p class="category-title">Lorem ipsum dolor sit amet</p>
      <div class="category-list d-flex flex-wrap mt-2">
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category One</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Two</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Three</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Four</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Five</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Six</span>
        </div>
        <div class="category border rounded my-1 mr-2 p-2">
          <span>Category Seven</span>
        </div>
      </div>
    </div>
  </main>
</template>

<script>
export default {
  name: 'CategoryCard',
  props: {
    hideCardClass: {
      type: Boolean,
      default: false,
    }
  }
}
</script>

<style lang="scss" scoped>
@import './CategoryCard'
</style>
