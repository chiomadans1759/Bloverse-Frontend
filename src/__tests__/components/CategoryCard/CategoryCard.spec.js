import { mount, RouterLinkStub  } from '@vue/test-utils'
import CategoryCard from '@/components/CategoryCard/CategoryCard.vue';

describe('CategoryCard Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(CategoryCard, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
