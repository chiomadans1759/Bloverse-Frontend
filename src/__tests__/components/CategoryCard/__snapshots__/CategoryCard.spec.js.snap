// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Component should render correctly 1`] = `
<main
  class="card category-card p-3"
>
  <div
    class="header d-flex justify-content-between "
  >
    <h5>
      <strong>
        Interest
      </strong>
    </h5>
     
    <span
      class="edit-icon"
    >
      <i
        class="fal fa-edit mr-2"
      />
    </span>
  </div>
   
  <div
    class="mt-1"
  >
    <p
      class="category-title"
    >
      Lorem ipsum dolor sit amet
    </p>
     
    <div
      class="category-list d-flex flex-wrap mt-2"
    >
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category One
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Two
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Three
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Four
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Five
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Six
        </span>
      </div>
       
      <div
        class="category border rounded my-1 mr-2 p-2"
      >
        <span>
          Category Seven
        </span>
      </div>
    </div>
  </div>
</main>
`;
